### PR TITLE
sink(cdc): adjust error in kafka producer adjust configuration logic.

### LINKED ISSUE #5293: Error messages in package kafka AdjustConfig() is misleading
### What did you do?

Set up a changefeed to an authorized kafka broker.

The error message is about `insync` and `replication-factor`.

```
[2022/04/26 15:13:45.080 +00:00] [INFO] [http_router.go:148] [/api/v1/changefeeds] [status=500] [method=POST] [path=/api/v1/changefeeds] [query=] [ip=[172.30.106.193](http://172.30.106.193/)] [user-agent=Go-http-client/1.1] [error="[CDC:ErrKafkaNewSaramaProducer]new sarama producer: [CDC:ErrKafkaInvalidConfig]because TiCDC Kafka producer's `request.required.acks` defaults to -1, TiCDC cannot deliver messages when the `replication-factor` is less than `min.insync.replicas`: kafka server: The client is not authorized to send this request type."] [errorVerbose="[CDC:ErrKafkaNewSaramaProducer]new sarama producer: [CDC:ErrKafkaInvalidConfig]because TiCDC Kafka producer's `request.required.acks` defaults to -1, TiCDC cannot deliver messages when the `replication-factor` is less than `min.insync.replicas`: kafka server: The client is not authorized to send this request type.\[ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/errors.WrapError\n\tgithub.com/pingcap/tiflow/pkg/errors/helper.go:30\ngithub.com/pingcap/tiflow/cdc/sink/producer/kafka.NewKafkaSaramaProducer\n\tgithub.com/pingcap/tiflow/cdc/sink/producer/kafka/kafka.go:284\ngithub.com/pingcap/tiflow/cdc/sink.newKafkaSaramaSink\n\tgithub.com/pingcap/tiflow/cdc/sink/mq.go:398\ngithub.com/pingcap/tiflow/cdc/sink.init.1.func3\n\tgithub.com/pingcap/tiflow/cdc/sink/sink.go:103\ngithub.com/pingcap/tiflow/cdc/sink.New\n\tgithub.com/pingcap/tiflow/cdc/sink/sink.go:123\ngithub.com/pingcap/tiflow/cdc/sink.Validate\n\tgithub.com/pingcap/tiflow/cdc/sink/sink.go:136\ngithub.com/pingcap/tiflow/cdc/capture.verifyCreateChangefeedConfig\n\tgithub.com/pingcap/tiflow/cdc/capture/http_validator.go:145\ngithub.com/pingcap/tiflow/cdc/capture.(*HTTPHandler).CreateChangefeed\n\tgithub.com/pingcap/tiflow/cdc/capture/http_handler.go:216\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\ngithub.com/pingcap/tiflow/cdc.errorHandleMiddleware.func1\n\tgithub.com/pingcap/tiflow/cdc/http_router.go:163\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\ngithub.com/pingcap/tiflow/cdc.timeoutMiddleware.func1\n\tgithub.com/pingcap/tiflow/cdc/http_router.go:126\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\ngithub.com/pingcap/tiflow/cdc.logMiddleware.func1\n\tgithub.com/pingcap/tiflow/cdc/http_router.go:135\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\ngithub.com/gin-gonic/gin.(*Engine).handleHTTPRequest\n\tgithub.com/gin-gonic/gin@v1.7.4/gin.go:489\ngithub.com/gin-gonic/gin.(*Engine).ServeHTTP\n\tgithub.com/gin-gonic/gin@v1.7.4/gin.go:445\nnet/http.serverHandler.ServeHTTP\n\tnet/http/server.go:2887\nnet/http.(*conn).serve\n\tnet/http/server.go:1952\nruntime.goexit\n\truntime/asm_amd64.s:1371](http://ngithub.com/pingcap/errors.AddStack/n/tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174/ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs/n/tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164/ngithub.com/pingcap/tiflow/pkg/errors.WrapError/n/tgithub.com/pingcap/tiflow/pkg/errors/helper.go:30/ngithub.com/pingcap/tiflow/cdc/sink/producer/kafka.NewKafkaSaramaProducer/n/tgithub.com/pingcap/tiflow/cdc/sink/producer/kafka/kafka.go:284/ngithub.com/pingcap/tiflow/cdc/sink.newKafkaSaramaSink/n/tgithub.com/pingcap/tiflow/cdc/sink/mq.go:398/ngithub.com/pingcap/tiflow/cdc/sink.init.1.func3/n/tgithub.com/pingcap/tiflow/cdc/sink/sink.go:103/ngithub.com/pingcap/tiflow/cdc/sink.New/n/tgithub.com/pingcap/tiflow/cdc/sink/sink.go:123/ngithub.com/pingcap/tiflow/cdc/sink.Validate/n/tgithub.com/pingcap/tiflow/cdc/sink/sink.go:136/ngithub.com/pingcap/tiflow/cdc/capture.verifyCreateChangefeedConfig/n/tgithub.com/pingcap/tiflow/cdc/capture/http_validator.go:145/ngithub.com/pingcap/tiflow/cdc/capture.(*HTTPHandler).CreateChangefeed/n/tgithub.com/pingcap/tiflow/cdc/capture/http_handler.go:216/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.7.4/context.go:165/ngithub.com/pingcap/tiflow/cdc.errorHandleMiddleware.func1/n/tgithub.com/pingcap/tiflow/cdc/http_router.go:163/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.7.4/context.go:165/ngithub.com/pingcap/tiflow/cdc.timeoutMiddleware.func1/n/tgithub.com/pingcap/tiflow/cdc/http_router.go:126/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.7.4/context.go:165/ngithub.com/pingcap/tiflow/cdc.logMiddleware.func1/n/tgithub.com/pingcap/tiflow/cdc/http_router.go:135/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.7.4/context.go:165/ngithub.com/gin-gonic/gin.(*Engine).handleHTTPRequest/n/tgithub.com/gin-gonic/gin@v1.7.4/gin.go:489/ngithub.com/gin-gonic/gin.(*Engine).ServeHTTP/n/tgithub.com/gin-gonic/gin@v1.7.4/gin.go:445/nnet/http.serverHandler.ServeHTTP/n/tnet/http/server.go:2887/nnet/http.(*conn).serve/n/tnet/http/server.go:1952/nruntime.goexit/n/truntime/asm_amd64.s:1371)"] [cost=344.444039ms]
```

### What did you expect to see?

Only shows the root cause `kafka server: The client is not authorized to send this request type`.

### What did you see instead?

See above.

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
v5.4.0
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #5293 

### What is changed and how it works?

* only report `replication-factor` and `min.insync.replicas` if check failed, do not wrap other errors.

* report `ErrKafkaNewSaramaProducer` when adjust the config`.

Screen Shot 2022-04-27 at 3 28 02 PM

### Check List 

Tests 

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by overvenus** [comment]
Could you test when the TiCDC is not authorized?

**Comment by 3AceShowHand** [comment]
> Could you test when the TiCDC is not authorized?

It looks quite hard to mock the scenario, I use `let broker report error` as a replacement in the test case.

**Comment by overvenus** [comment]
@3AceShowHand Could you check other error messages in kafka.go?

**Review by overvenus — APPROVED** [review]

File: cdc/sink/mq/producer/kafka/config.go [hunk 1]
```diff
@@ -75,6 +75,8 @@ func (c *Config) setPartitionNum(realPartitionCount int32) error {
 	// user does not specify the `partition-num` in the sink-uri
 	if c.PartitionNum == 0 {
 		c.PartitionNum = realPartitionCount
+		log.Info("partitionNum is not set, set by topic's partition-num",
+			zap.Int32("partition-num", realPartitionCount))
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/sink/mq/producer/kafka/config.go:79]: Oh I see, the regex doesn't match `Int32`
```diff
--- a/scripts/check-log-style.sh
+++ b/scripts/check-log-style.sh
@@ -12,7 +12,7 @@
 # limitations under the License.

 # zap field name should be camelCase, excepts for idioms and special terms.
-grep -RnE "zap.[A-Z][a-z]+\(\"[0-9A-Za-z]*[-_ ][0-9A-Za-z]*\"(,|\))" cdc tests pkg |
+grep -RnE "zap.[A-Z][a-z][0-9]+\(\"[0-9A-Za-z]*[-_ ][0-9A-Za-z]*\"(,|\))" cdc tests pkg |
```

[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/sink/mq/producer/kafka/config.go:79]: FYI filed a PR in https://github.com/pingcap/tiflow/pull/5323

**Review by sdojjy — APPROVED** [review]

**Pull request merged**